src/Microsoft.Data/PrimitiveColumn.cs

```diff
@@ -196,6 +230,31 @@ internal PrimitiveColumn<decimal> CloneAsDecimalColumn()
             return new PrimitiveColumn<decimal>(Name, newColumnContainer);
         }
 
+        public override Dictionary<TKey, ICollection<long>> HashColumnValues<TKey>()
+        {
+            if (typeof(TKey) == typeof(T))
+            {
+                Dictionary<T, ICollection<long>> multimap = new Dictionary<T, ICollection<long>>(EqualityComparer<T>.Default);
+                for (long i = 0; i < Length; i++)
+                {
+                    bool containsKey = multimap.TryGetValue(this[i] ?? default, out ICollection<long> values);
+                    if (containsKey)
+                    {
+                        values.Add(i);
+                    }
+                    else
+                    {
+                        multimap.Add(this[i] ?? default, new List<long>() { i });
```

Comment:
This would mean `Null` values in primitive columns will be equivalent to `0`. I *think* this is ok

Reply:
Filed https://github.com/dotnet/corefxlab/issues/2700 to address this. `null` should be it's own group